### DISCUSSION TITLE
Rendering nested data

### DISCUSSION THREAD

**Discussion by hussamkhatib** [nick-keller/react-datasheet-grid#280] [Q&A] [answered]
Is there a way to render nested data .

Coinsider an example
```
export const columns: Column[] = [
    {
        ...keyColumn('id', textColumn),
        title: 'ID',
    },
    {
        ...keyColumn('name', textColumn),
        title: 'Name',
    },
     ..... 
  ];
```

the data coming from server might look something like this

```
type Value = {
   id: string
   user: {
      name: string;
      age: numbere;
   }
}
```

Usage
```
  <DataSheetGrid
    value={data}
    columns={columns}
/>
```

There are two solutions which I could think of, which does seem to be the ideal way fo doing it.
- Custom components 
- Transforming the `value` into a flat array before passing in the Grid

Note: I dont want them to render them in a nested way in the UI. I want them to render just like it is coming from flat array, just that I will be getting response in nested data from server

**Comment by nick-keller** [comment, marked as answer]
Hello @hussamkhatib ,
I would suggest to flatten the data you feed to the component.
In your state you would have a flattened version of your data, and only transform it when you first load the data, and when you send it back to the server.
